includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -717,6 +721,54 @@ public function prepare_order_source( $order = null ) {
 		);
 	}
 
+	/**
+	 * Retrieves a source based on an order. Checks for the customer, associated
+	 * with the order and/or the customer which was manually entered in the admin.
+	 *
+	 * @param WC_Order $order The order to use as a starting point.
+	 * @return stdClass|null  Either an object with the necessary data or null.
+	 */
+	public function get_default_customer_source_for_order( $order ) {
+		$stripe_customer = null;
+		$order_id        = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $order->id : $order->get_id();
+
+		// Check for explicit manual input first.
+		$stripe_customer_id = get_post_meta( $order_id, '_stripe_customer_id', true );
```

Comment:
Some of the steps we take in `get_default_customer_source_for_order`, like looking up the `$stripe_customer_id` are already done in `prepare_order_source`:

https://github.com/woocommerce/woocommerce-gateway-stripe/blob/741225ad9a56bdb48c07d0749d64f51004ed1911/includes/abstracts/abstract-wc-stripe-payment-gateway.php#L680

There's some room to minimize those.

Reply:
The line from the example seems to be the only duplicate from `prepare_order_source`, but it also happens to be needed for when the method is called from within `validate_subscription_payment_meta` methods. I think keeping it within the abstract class is the better option.